lib/BBTools/utils/BBMapRunner.py

```diff
@@ -196,24 +201,78 @@ def _process_app_params_to_cli(self, io_params, app_params, output_dir, run_log,
+                if int(app_params['require_correct_strand']) != 0:
+                    if not app_params.get('input_parameter_suite') \
+                       or 'rcs=' not in app_params.get('input_parameter_suite'):
+                        options.append("rcs=t")
+                else:
+                    if not app_params.get('input_parameter_suite') \
+                       or 'rcs=' not in app_params.get('input_parameter_suite'):
+                        options.append("rcs=f")
+
+            if app_params.get('min_id') and float(app_params['min_id']) != 0.0:
+                if not app_params.get('input_parameter_suite') \
+                   or 'min=' not in app_params.get('input_parameter_suite'):
+                    options.append("minid={}".format(str(app_params['min_id'])))
 
-        # use pigz and unpigz for what are usually huge files
-        options.append('pigz=t')
-        options.append('unpigz=t')
+            if app_params.get('speed_mode'):
+                if app_params['speed_mode'] != 'default':
+                    if 'slow' not in app_params.get('input_parameter_suite') \
+                       and 'fast' not in app_params.get('input_parameter_suite'):
+                        options.append(app_params['speed_mode'])
 
-        # enforce qual=33 if not given
-        if app_params.get('qual_score_mode'):
-           options.append('qin={}'.format(str(app_params['qual_score_mode'])))
-        else:
-           options.append('qin={}'.format('33'))
+            if app_params.get('subfilter_thresh'):
+                if not app_params.get('input_parameter_suite') \
+                   or 'subfilter=' not in app_params.get('input_parameter_suite'):
+                    options.append("subfilter={}".format(str(app_params['subfilter_thresh'])))
+
+            if app_params.get('delfilter_thresh'):
+                if not app_params.get('input_parameter_suite') \
+                   or 'delfilter=' not in app_params.get('input_parameter_suite'):
+                    options.append("delfilter={}".format(str(app_params['delfilter_thresh'])))
+
+            # hard-code max threads for now
+            options.append('t={}'.format('4'))
+
+            # use pigz and unpigz for what are usually huge files
+            options.append('pigz=t')
+            options.append('unpigz=t')
+
+            # enforce qual=33 if not given
+            if app_params.get('qual_score_mode'):
+                options.append('qin={}'.format(str(app_params['qual_score_mode'])))
+            else:
+                options.append('qin={}'.format('33'))
 
-        # add the memory requirement at the end
-        options.append('-Xmx{}g'.format(mem))
+            # add the memory requirement at the end
+            options.append('-Xmx{}g'.format(mem))
+
+            # finally, route stderr (a log file) to a file in the output dir
+            options = options + ['2>', run_log]
+
+            # enable multi assemblies later
+            break
```

Comment:
Having the `break` here just means you're processing the first assembly in the list, right? Just checking.

Reply:
right.  temporary until I add support for a list of more than one assembly targets working.

Reply:
👍